src/test/regress/sql/window_functions.sql

```diff
@@ -270,6 +269,69 @@ WINDOW
 ORDER BY
 	user_id, value_1, 3, 4;
 
+-- repeat above 3 tests without grouping by distribution column
+SELECT
+	value_2,
+	rank() OVER (PARTITION BY value_2 ROWS BETWEEN
+				 UNBOUNDED PRECEDING AND CURRENT ROW),
+	dense_rank() OVER (PARTITION BY value_2 RANGE BETWEEN
+					   UNBOUNDED PRECEDING AND CURRENT ROW),
+	CUME_DIST() OVER (PARTITION BY value_2 RANGE BETWEEN
+					  UNBOUNDED PRECEDING AND  UNBOUNDED FOLLOWING),
+	PERCENT_RANK() OVER (PARTITION BY value_2 ORDER BY avg(value_1) RANGE BETWEEN
```

Comment:
In the past, we had weird bugs around `PARTITION BY avg(col)` followed by `ORDER BY avg(col)` in the pushdown case.

Might be good to add a test (and compare the result with postgres) with `order by` in the window clause and/or the query itself

(I didn't see any, if we already have please ignore this comment)

Reply:
I assume you mean to average by two different columns as ordering by the partitioning expression is meaningless. Added test with `dense_rank() OVER (PARTITION BY avg(value_3) ORDER BY avg(value_2))`